### PR TITLE
[docs] Add docs for array notated hook callbacks

### PR DESCRIPTION
**Pull request by andrewnicols**
Docs for MDL-81180

### CODE REVIEW

File: docs/apis/core/hooks/index.md
Comment (mickhawkins) [inline docs/apis/core/hooks/index.md:1]: [Line 215](https://github.com/moodle/devdocs/pull/915/commits/fdaeb718f103303e79952ed0c16781e80b713e7e#diff-741308c9e41acedf977da265aa2c22f67abdec44fc559449ff1121a7844aa290R215) is missing the required end of line comma.

Comment (mickhawkins) [inline docs/apis/core/hooks/index.md:1]: [Line 238](https://github.com/moodle/devdocs/pull/915/commits/fdaeb718f103303e79952ed0c16781e80b713e7e#diff-741308c9e41acedf977da265aa2c22f67abdec44fc559449ff1121a7844aa290R238) minor typo - double space before "either".

File: docs/devupdate.md
Comment (mickhawkins) [inline docs/devupdate.md:1]: [Line 64](https://github.com/moodle/devdocs/pull/915/commits/fdaeb718f103303e79952ed0c16781e80b713e7e#diff-fbaad1f7aff998327ea762fc02976ac1a8c3e158abe6e49571ff71c2b8d6dc08R64) looks like the label ("array notation") seems incorrect. Should that be "Discouraged string notation" or something along those lines?

**Review by mickhawkins — CHANGES_REQUESTED** [review]
Hi @andrewnicols I just found a few minor errors, can you please address those?

File: docs/devupdate.md
Comment (mickhawkins) [reply docs/devupdate.md:1]: Forgot to mention this is also flagged as having a conflict since you pushed it.

**Comment by andrewnicols** [comment]
Thanks Mick,

All addresed.

**Review by mickhawkins — APPROVED** [review]
Thanks Andrew, confirmed those are all addressed.

**Pull request merged**